src/rngs/small.rs

```diff
@@ -10,30 +10,32 @@
 
 use rand_core::{Error, RngCore, SeedableRng};
 
-#[cfg(all(not(target_os = "emscripten"), target_pointer_width = "64"))]
-type Rng = rand_pcg::Pcg64Mcg;
-#[cfg(not(all(not(target_os = "emscripten"), target_pointer_width = "64")))]
-type Rng = rand_pcg::Pcg32;
+#[cfg(target_pointer_width = "64")]
+type Rng = super::xoshiro256plusplus::Xoshiro256PlusPlus;
+#[cfg(not(target_pointer_width = "64"))]
+type Rng = super::xoshiro128plusplus::Xoshiro128PlusPlus;
 
 /// A small-state, fast non-crypto PRNG
 ///
 /// `SmallRng` may be a good choice when a PRNG with small state, cheap
 /// initialization, good statistical quality and good performance are required.
-/// It is **not** a good choice when security against prediction or
-/// reproducibility are important.
+/// It is **not** a good choice when:
+/// - Security against prediction or reproducibility are important.
+///   Use [`StdRng`] instead.
+/// - The best runtime performance on platforms with SIMD support is required.
+///   Use [`StdRng`] or `rand_chacha::ChaCha8Rng` instead.
```

Comment:
In general this will depend heavily on the application (bytes vs u32 vs u64 usage, interactions with the cache, whether registers can be reserved by the RNG across multiple operations). The only consistent performance advice we can give is to benchmark ones own application.

Maybe it's worth also pointing out that when low quality keys are used, either some type of key mixing is required or one should prefer crypto-style RNGs?

Reply:
I revised the advise. What do you think?